.travis.yml

```diff
@@ -1,34 +1,77 @@
 language: rust
 rust:
   - stable
-cache: cargo
-# necessary for `travis-cargo coveralls --no-sudo`
+
+sudo: false
+git:
+  depth: 3 # We don't need to clone with depth 50
+
+cache:
+  - cargo
+  - pip
+
 addons:
   apt:
     packages:
       - libcurl4-openssl-dev
       - libelf-dev
       - libdw-dev
       - binutils-dev
+      - lua5.2
+
 notifications:
   slack:
     rooms:
       - way-cooler:W2vOSvJvgxGi7EiGkZCtvlB6#way-cooler
-    on_success: :change
-    on_failure: :change
+    on_success: change
+    on_failure: change
   email: false
-# load travis-cargo
-before_script:
-  - |
-      git clone https://github.com/Timidger/travis-cargo
-      (cd travis-cargo;
-      pip install -e . --user &&
-      export PATH=$HOME/.local/bin:$PATH)
+
+before_install:
+  - git clone --depth=3 https://github.com/Timidger/travis-cargo
+  - export PATH=$HOME/.local/bin:$PATH
+
+install:
+  # install travis-cargo
+  - pip install --user -e ./travis-cargo
+  # install dependencies for ci.py
+  - pip install --user docopt
+  # Sometimes cargo check doesn't exist?
+  - cargo install --force cargo-check
+
 script:
+  # check integrity of Lua files in /lib/lua/
+  - for file in $( find lib/lua -type f -name "*.lua" ); do echo Checking $file; luac -p $file; done
+  # check integrity of init file
+  - luac -p config/init.lua
+  # Make sure no-test builds
+  - cargo check --verbose
+  # run tests
   - cargo test --verbose
+  # run tag checks
+  - python $TRAVIS_BUILD_DIR/ci.py travis-check
 
 after_success:
-  # measure code coverage and upload to coveralls.io (the verify
-  # argument mitigates kcov crashes due to malformed debuginfo, at the
-  # cost of some speed <https://github.com/huonw/travis-cargo/issues/12>)
+  # Test and upload coverage to coveralls
+  # Justification of excluded files:
+  # main.rs: just initialization of logging, sigkill handler, and initializing lazy static objects. (Arg parsing should be moved into a different module and tested)
+  # mod.rs: although at the moment we have too much code in various mod.rs's these are not really testable either way
+  # tests.rs: should not be running on our tests themselves
+  # wayland_client_api.rs: we really don't have a good way of testing any of this. At the moment it's simple enough that if it didn't work it would be obvious in user testing.
+  # callbacks.rs: we don't have any integration tests for callbacks right now. They'd need to deal with the lazy statics that the other components of the program live in.
   - travis-cargo coveralls --no-sudo --verify --exclude-pattern="main.rs,mod.rs,tests.rs,wayland_client_api.rs,callbacks.rs"
+
+before_deploy:
+  - ci.py prepare-deploy
+deploy:
+  provider: releases
+  skip_cleanup: false # We can skip cleanup as we recompile with --release
```

Comment:
Wait... is this skipping clean up or not? The comment seems to indicate we can, because we recompile with --release, but `skip_cleanup` is set to `false`. Or is this correct and the option is just reading confusingly to me.